edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/TwinManager.cs

```diff
@@ -604,6 +604,36 @@ static void ValidateTwinProperties(JToken properties, int currentDepth)
             }
         }
 
+        // TODO: Move to a Twin helper class (along with Twin manager update).
+        internal static string EncodeTwinKey(string key)
+        {
+            Preconditions.CheckNonWhiteSpace(key, nameof(key));
+            var sb = new StringBuilder();
+            foreach (char ch in key)
+            {
+                switch (ch)
+                {
+                    case '.':
+                        sb.Append("%2E");
+                        break;
+
+                    case '$':
+                        sb.Append("%24");
+                        break;
+
+                    case ' ':
+                        sb.Append("%20");
+                        break;
+
+                    default:
+                        sb.Append(ch);
+                        break;
```

Comment:
Super nit: Not needed.

Reply:
Actually interestingly enough, it is needed..